Makefile

```diff
@@ -1,41 +1,57 @@
-# Common to all versions of K
-# ===========================
+# Settings
+# --------
+
+BUILD_DIR:=$(CURDIR)/.build
+K_SUBMODULE:=$(BUILD_DIR)/k
+PANDOC_TANGLE_SUBMODULE:=$(BUILD_DIR)/pandoc-tangle
+BUILD_LOCAL:=$(BUILD_DIR)/local
+
+PKG_CONFIG_PATH:=$(BUILD_LOCAL)/lib/pkgconfig
+export PKG_CONFIG_PATH
+
+LUA_PATH:=$(PANDOC_TANGLE_SUBMODULE)/?.lua;;
+export LUA_PATH
 
-.PHONY: all clean deps k-deps ocaml-deps build defn sphinx split-tests \
+.PHONY: all clean deps k-deps tangle-deps ocaml-deps build defn sphinx split-tests \
 		test test-all vm-test vm-test-all bchain-test bchain-test-all proof-test proof-test-all
 
 all: build split-tests
 
 clean:
-	rm -rf .build/java .build/ocaml .build/node .build/logs tests/proofs .build/k/make.timestamp .build/local
+	rm -rf .build/java .build/ocaml .build/node .build/logs tests/proofs .build/k/make.timestamp .build/pandoc-tangle/make.timestamp .build/local
 
-build: .build/ocaml/driver-kompiled/interpreter .build/java/driver-kompiled/timestamp
+distclean: clean
+	opam switch system
+	opam switch remove 4.03.0+k --yes || true
+	git submodule deinit --force -- ./
```

Comment:
does this remove untracked files in the submodules, though? if not, you will still want to do a `mvn clean` in k before you deinit

Reply:
https://stackoverflow.com/questions/35940627/git-how-to-remove-untracked-folder-in-submodules

looks like it doesn't. Would you prefer `cd $(K_SUBMODULE) && mvn clean -q`, or `cd $(K_SUBMODULE) && git reset --hard`? I'm leaning towards the first, or we could do both.

Reply:
mvn clean would be better for a clean target.